python/kserve/kserve/protocol/dataplane.py

```diff
@@ -204,7 +204,7 @@ def model_ready(self, model_name: str) -> bool:
 
         return self._model_registry.is_model_ready(model_name)
 
-    def decode(self, body, headers) -> Tuple[Union[Dict, InferRequest, CloudEvent], Dict]:
+    def decode(self, body, headers) -> Tuple[Union[Dict, InferRequest], Dict]:
```

Comment:
@yuzisun I think we should keep the cloudevent. Because In `avro` scenario we are not decoding the cloudevent.

Reply:
Though it is no longer type of CloudEvent, it is a dict with the Avro payload right ?

Reply:
If you see the code block here, we are only decoding it if the content type is json. If the content type is  avro it just returns the raw cloud event.  https://github.com/kserve/kserve/blob/0c96effc0834b0b1ba91d4dd39a125d8660dd74c/python/kserve/kserve/protocol/dataplane.py#LL229C8-L242C1